packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageViewManager.kt

```diff
@@ -0,0 +1,132 @@
+package expo.modules.image
+
+import com.bumptech.glide.Glide
+import com.bumptech.glide.RequestManager
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReadableMap
+import com.facebook.react.uimanager.PixelUtil
+import com.facebook.react.uimanager.SimpleViewManager
+import com.facebook.react.uimanager.Spacing
+import com.facebook.react.uimanager.ThemedReactContext
+import com.facebook.react.uimanager.ViewProps
+import com.facebook.react.uimanager.annotations.ReactProp
+import com.facebook.react.uimanager.annotations.ReactPropGroup
+import com.facebook.yoga.YogaConstants
+import expo.modules.image.enums.ImageResizeMode
+import expo.modules.image.events.ImageErrorEvent
+import expo.modules.image.events.ImageLoadEvent
+import expo.modules.image.events.ImageLoadStartEvent
+import expo.modules.image.events.ImageProgressEvent
+import expo.modules.image.okhttp.OkHttpClientProgressInterceptor
+
+private val BORDER_LOCATIONS = intArrayOf(
+  Spacing.ALL,
+  Spacing.LEFT,
+  Spacing.RIGHT,
+  Spacing.TOP,
+  Spacing.BOTTOM,
+  Spacing.START,
+  Spacing.END
+)
+
+class ExpoImageViewManager(applicationContext: ReactApplicationContext) : SimpleViewManager<ExpoImageView>() {
+  private val mRequestManager: RequestManager = Glide.with(applicationContext)
+  private val mProgressInterceptor: OkHttpClientProgressInterceptor = OkHttpClientProgressInterceptor
+  override fun getName() = "ExpoImage"
+
+  override fun getExportedCustomDirectEventTypeConstants() =
+    mutableMapOf(
+      ImageLoadStartEvent.EVENT_NAME to mutableMapOf("registrationName" to ImageLoadStartEvent.EVENT_NAME),
+      ImageProgressEvent.EVENT_NAME to mutableMapOf("registrationName" to ImageProgressEvent.EVENT_NAME),
+      ImageErrorEvent.EVENT_NAME to mutableMapOf("registrationName" to ImageErrorEvent.EVENT_NAME),
+      ImageLoadEvent.EVENT_NAME to mutableMapOf("registrationName" to ImageLoadEvent.EVENT_NAME)
+    )
+
+  // Props setters
+  @ReactProp(name = "source")
+  fun setSource(view: ExpoImageView, sourceMap: ReadableMap?) {
+    view.sourceMap = sourceMap
+  }
+
+  @ReactProp(name = "resizeMode")
+  fun setResizeMode(view: ExpoImageView, stringValue: String) {
+    val resizeMode = ImageResizeMode.fromStringValue(stringValue)
+    view.setResizeMode(resizeMode)
+  }
+
+  @ReactPropGroup(names = [
+    ViewProps.BORDER_RADIUS,
+    ViewProps.BORDER_TOP_LEFT_RADIUS,
+    ViewProps.BORDER_TOP_RIGHT_RADIUS,
+    ViewProps.BORDER_BOTTOM_RIGHT_RADIUS,
+    ViewProps.BORDER_BOTTOM_LEFT_RADIUS,
+    ViewProps.BORDER_TOP_START_RADIUS,
+    ViewProps.BORDER_TOP_END_RADIUS,
+    ViewProps.BORDER_BOTTOM_START_RADIUS,
+    ViewProps.BORDER_BOTTOM_END_RADIUS
+  ], defaultFloat = YogaConstants.UNDEFINED)
+  fun setBorderRadius(view: ExpoImageView, index: Int, borderRadius: Float) {
+    val radius = if (!YogaConstants.isUndefined(borderRadius) && borderRadius < 0) YogaConstants.UNDEFINED else borderRadius
```

Comment:
`normalizeYogaSize`, `normalizeYogaPositive`, `clampYogaPositive`?

but since `YogaConstants.UNDEFINED` evaluates to `NaN`, maybe we can omit the word `Yoga` and use `normalizeToPositive` or `clampToPositive`?

Reply:
I kept Yoga in names I think if we use methods from YogaContants we shouldn't hide this fact.